shopify-client/src/views/ProductUpdate.vue

```diff
@@ -0,0 +1,70 @@
+<template>
+  <div class="center">
+    <p v-if="error">{{ error }}</p>
+    <ProductForm
+      v-if="product"
+      v-bind:handleSubmit="handleSubmit"
+      v-bind:initialValues="product"
+    />
+  </div>
+</template>
+
+<script>
+import ProductForm from "@/components/ProductForm.vue";
+
+export default {
+  name: "ProductUpdate",
+  props: ["id"],
+  components: {
+    ProductForm,
+  },
+  data: function() {
+    return {
+      product: null,
+      error: null,
+    };
+  },
+  created: function() {
+    const vm = this;
+    fetch(`/api/products/${vm.id}`)
+      .then((response) => response.json())
+      .then((response) => {
+        if (response.status == 404) {
+          throw new Error(response.message);
+        }
+
+        return response;
+      })
+      .then((data) => (vm.product = data))
+      .catch((error) => {
+        vm.error = error;
```

Comment:
Question: this error will contain a huge json containing the full stack trace, do we want to show this on the client?

Reply:
The error message should only say "Product with <id> not found". Have you tried it?